hail/src/test/scala/is/hail/annotations/RegionSuite.scala

```diff
@@ -232,4 +235,33 @@ class RegionSuite extends TestNGSuite {
       assert(pool.numFreeBlocks() == 2)
     }
   }
+
+  @Test
+  def testChunkCache(): Unit = {
+    RegionPool.scoped { pool =>
+
+      val operations = ArrayBuffer[(String, Long, Long)]()
+
+      def allocate(numBytes: Long): Long = {
+        val pointer = Memory.malloc(numBytes)
+        operations += (("allocate", numBytes, pointer))
+        pointer
+      }
+      def free(ptrToFree: Long): Unit = {
+        operations += (("free", ptrToFree, 0L))
+        Memory.free(ptrToFree)
+      }
+      val chunkCache = new ChunkCache(allocate, free)
+      val sizeArr = IndexedSeq(30L, 60L, 500L, 300L, 28L, 58L, 90L, 300L, 20L, 480L)
+      val ab = new LongArrayBuilder()
+      var i = 0
+      while (i <= 9) {
+        ab += chunkCache.getChunk(pool, sizeArr(i))._1
+        i += 1
+      }
+      chunkCache.freeChunksToCache(ab)
+      chunkCache.freeAll(pool)
+
+    }
+  }
```

Comment:
We talked in person about what this test should be. We want to look at the operations arrays to check that we use the cache when appropriate, and free when appropriate.